src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -221,12 +221,36 @@ class CompositeComparisonOperatorSegment(BaseSegment):
         StringLexer("vertical_bar", "|", CodeSegment),
         StringLexer("caret", "^", CodeSegment),
         StringLexer("star", "*", CodeSegment),
-        StringLexer("bracket_open", "(", CodeSegment),
-        StringLexer("bracket_close", ")", CodeSegment),
-        StringLexer("sq_bracket_open", "[", CodeSegment),
-        StringLexer("sq_bracket_close", "]", CodeSegment),
-        StringLexer("crly_bracket_open", "{", CodeSegment),
-        StringLexer("crly_bracket_close", "}", CodeSegment),
+        StringLexer(
+            "start_bracket", "(", CodeSegment, segment_kwargs={"type": "start_bracket"}
+        ),
```

Comment:
It's a good question. At the moment, the `name` of the lexer isn't automatically turned into the `type` of the segment. It's something we might do in future, but not currently automatic - because it changes a lot of the tests. Definitely something to think about in future though 🤔